src/addons/transitions/ReactCSSTransitionGroupChild.js

```diff
@@ -48,8 +48,8 @@ var ReactCSSTransitionGroupChild = React.createClass({
 
   transition: function(animationType, finishCallback) {
     var node = React.findDOMNode(this);
-    var className = this.props.name + '-' + animationType;
-    var activeClassName = className + '-active';
+    var className = this.props.name[animationType] || this.props.name + '-' + animationType;
+    var activeClassName = this.props.name[animationType + 'Active'] || className + '-active';
```

Comment:
@zpao can you help on this? I would also love to help out on this if i can. Maybe with the test?